### PR TITLE
PHP 8.1 compatibility updates

### PR DESCRIPTION
**Pull request by markbiek**
See: https://github.com/Automattic/wp-calypso/issues/83721

### What and why? 🤔

This makes some minor updates for PHP 8.1 compatibility. This is part of a larger Automattic initiative to get the WPCOM codebase and all libraries used ready for migrating to PHP 8.x.

More discussion in p1698769503301209-slack-C05MFVC6L74

### Testing Steps ✍️

- Visual inspection of changes
- Run unit tests

### You're it! 👑

@sanmai

### CODE REVIEW

**Review by sixhours — APPROVED** [review]
LGTM!

File: extras/phpcs/PHPCSDiff.php [hunk 1]
```diff
@@ -331,10 +331,10 @@ private function filterMessages(array $messages, array $added_lines): array
     protected function getErrors(array $data, string $type = 'ERROR'): array
     {
         $errors = [];
-        if (!isset($data['files']) || count($data['files']) < 1) {
+        if (!isset($data['files']) || !is_array($data['files']) || count($data['files']) < 1) {
             return [];
         }
-        if (count($data['files']) > 1) {
+        if (is_array($data['files']) && count($data['files']) > 1) {
```
Comment (cyle) [inline extras/phpcs/PHPCSDiff.php]: we would never get to the condition where `$data['files']` isn't an array, thanks to the change on line 334 :eyes:

File: extras/phpcs/TumblrApp/Sniffs/Formatting/CurlynewlineSniff.php [hunk 2]
```diff
@@ -52,7 +52,8 @@ public function process(\PHP_CodeSniffer\Files\File $phpcsFile, $stackPtr)
 
         // Ensure we don't have an empty class/method
         $has_body = false;
-        for ($i = $curly+1; $i < count($tokens); $i++) {
+		$token_count = is_countable($tokens) ? count($tokens) : 0;
+        for ($i = $curly+1; $i < $token_count; $i++) {
```
Comment (cyle) [inline extras/phpcs/TumblrApp/Sniffs/Formatting/CurlynewlineSniff.php]:
```suggestion
        for ($i = $curly + 1; $i < $token_count; $i++) {
```

while you're here 😄

File: extras/phpcs/PHPCSDiff.php (hunk 1 above)
Comment (markbiek) [reply extras/phpcs/PHPCSDiff.php]: Ah, yes. Good catch!

Comment (cyle) [reply extras/phpcs/PHPCSDiff.php]:
```suggestion
        if (count($data['files']) > 1) {
```

so can we leave it as-is in that case?

Comment (markbiek) [reply extras/phpcs/PHPCSDiff.php]: Yeah. I tried to leave it as-is but I messed up the rebase.

**Review by cyle — APPROVED** [review]

**Comment by markbiek** [comment]
@cyle Thanks for the review! Do you have any sense of when we might be able to push out a new version of the library?

**Comment by cyle** [comment]
> @cyle Thanks for the review! Do you have any sense of when we might be able to push out a new version of the library?

i'm not sure -- i'm going to be AFK for the next several days, so hopefully @sanmai and/or @lucila can help you with that part!

**Review by sanmai — COMMENTED** [review]
Changes make sense :+1:

**Comment by markbiek** [comment]
@sanmai @lucila I would be happy to take care of publishing a new release but I'm not sure what the process is. Can you all give me some guidance or would you rather do the release?

**Comment by sanmai** [comment]
It is done: https://github.com/Automattic/stream-builder/releases/tag/0.9.2

I think you should have received a ping since you are mentioned there.

**Comment by markbiek** [comment]
> It is done: https://github.com/Automattic/stream-builder/releases/tag/0.9.2
> 
> I think you should have received a ping since you are mentioned there.

Strange, I didn't get a ping. Thanks for getting that released so quickly!

**Pull request merged**